### ISSUE TITLE
PHP7 (and HHVM??) In travis

### ISSUE THREAD

**Issue by fntlnz** [facile-it/paraunit#1]
Since PHP 7 is in the process of being released IMHO we should consider start supporting it right now.

A first step to do this could be to tell travis to build the project against it as described [here](http://docs.travis-ci.com/user/languages/php/)

Are we considering to support this under HHVM?

**Comment by peelandsee** [comment]
+1 for this feature

**Comment by ranpafin** [comment]
We would gladly accept a pull request for this feature.

**Comment by taueres** [comment]
Porting to Hacklang would be awesome!

**Comment by fntlnz** [comment]
For this purpose I ran some commands against PHP 7.0.0beta1.

I used a docker image of mine [fntlnz/php-fpm:7.0.0beta1](https://registry.hub.docker.com/u/fntlnz/php-fpm/)

## paraunit cli

**COMMAND**
```
docker run -it -v $PWD:/paraunit -w /paraunit --entrypoint=php fntlnz/php-fpm:7.0.0beta1 /paraunit/src/Paraunit/Bin/paraunit run
```

**OUTPUT**
```
                                                   B>                           
                                                   B "Bp                        
.pp..                                              B    9p                      
 "9BBBBBBpp.                                       B      9p                    
    " ""9BBBBBBpp                          .<eeP"B B      .B b                  
           "SANDROBpp              .     B B     B B      )B B                  
              "BFRABBBB>  .<pe6P\B B     B B     B B      $  B     .e           
                 5NICOBBB B     ·B B     B B     B Bqp.  :B  B     $ 4BBpp      
                   BMIKIB B        B     B B     B B   "^Bp  B    ) |BBB"\BBpp. 
                 .BALEBBB """9q.   B"""""B B"""""B B      1p B""""9p BBBBbBBBBBBB
               <BLUCABBBB B    "B  B     B B     B B       B B     9 9BBB< ^P"  
            .6BSERGIOBBBB B666666B B     B B     B B       9 P      7 9BBBBP    

PARAUNIT TEST 0.4 -- by Shark Dev Team @ Facile.it
.................S

Execution time -- 00:00:00 

Executed: 5 tests

1 files with SKIPPED:
 RunnerTest.php
```

## paraunit testsuite

```
docker run -it -v $PWD:/paraunit -w /paraunit --entrypoint=php fntlnz/php-fpm:7.0.0alpha2 ./bin/phpunit
```

```
PHPUnit 4.8-dev by Sebastian Bergmann and contributors.

..............................S

Time: 374 ms, Memory: 6.00Mb

OK, but incomplete, skipped, or risky tests!
Tests: 31, Assertions: 58, Skipped: 1.
```

## zendframework/zend-diactoros testsuite

This command can run any zf component, just replace `zend-diactoros` with `zend-anythingyouwant`

**COMMAND**
```
docker run -it --entrypoint=bash fntlnz/php-fpm:7.0.0alpha2 -c "mkdir -p ~/.ssh; echo -e 'Host github.com\n\tStrictHostKeyChecking no\n' >> ~/.ssh/config; yum install git -y; git clone https://github.com/zendframework/zend-diactoros.git repository; cd repository; wget -O composer.phar https://getcomposer.org/composer.phar; chmod +x composer.phar; ./composer.phar require facile-it/paraunit:dev-master; ./composer.phar install; ./vendor/bin/paraunit run"
```

**OUTPUT**

```
                                                   B>                           
                                                   B "Bp                        
.pp..                                              B    9p                      
 "9BBBBBBpp.                                       B      9p                    
    " ""9BBBBBBpp                          .<eeP"B B      .B b                  
           "SANDROBpp              .     B B     B B      )B B                  
              "BFRABBBB>  .<pe6P\B B     B B     B B      $  B     .e           
                 5NICOBBB B     ·B B     B B     B Bqp.  :B  B     $ 4BBpp      
                   BMIKIB B        B     B B     B B   "^Bp  B    ) |BBB"\BBpp. 
                 .BALEBBB """9q.   B"""""B B"""""B B      1p B""""9p BBBBbBBBBBBB
               <BLUCABBBB B    "B  B     B B     B B       B B     9 9BBB< ^P"  
            .6BSERGIOBBBB B666666B B     B B     B B       9 P      7 9BBBBP    

PARAUNIT TEST 0.4 -- by Shark Dev Team @ Facile.it
................................................................................
................................................................................
................................................................................
................................................................................
................................................................................
............

Execution time -- 00:00:01 

Executed: 14 tests
```

**Comment by Jean85** [comment]
Thanks for the Docker one-line command! Very useful!
As you can see, it runs on PHP7 and Hacklang, but it doesn't run all the tests that PHPUnit runs normally. That's the major issue we're dealing with right now.

BTW, the `Executed: 14 tests` means 14 CLASSES, we will add a test function count (like PHPUnit) ASAP for fast comparison.

**Comment by Jean85** [comment]
This seems to be fixed with the latest commit (75cfc1f2760b180d645b78b9f7954cdd6cbdd23a):
```
                                                   B>                           
                                                   B "Bp                        
.pp..                                              B    9p                      
 "9BBBBBBpp.                                       B      9p                    
    " ""9BBBBBBpp                          .<eeP"B B      .B b                  
           "SANDROBpp              .     B B     B B      )B B                  
              "BFRABBBB>  .<pe6P\B B     B B     B B      $  B     .e           
                 5NICOBBB B     ·B B     B B     B Bqp.  :B  B     $ 4BBpp      
                   BMIKIB B        B     B B     B B   "^Bp  B    ) |BBB"\BBpp. 
                 .BALEBBB """9q.   B"""""B B"""""B B      1p B""""9p BBBBbBBBBBBB
               <BLUCABBBB B    "B  B     B B     B B       B B     9 9BBB< ^P"  
            .6BSERGIOBBBB B666666B B     B B     B B       9 P      7 9BBBBP    

PARAUNIT TEST 0.4 -- by Shark Dev Team @ Facile.it
................................................................................
................................................................................
................................................................................
................................................................................
................................................................................
................................................................................
................................................................................
...............................................................

Execution time -- 00:00:01 

Executed: 19 tests
```
If you count the dots, they are 623, and it's correct. The test counted on the bottom are still classes, I will address this ASAP, as I said before.

**Comment by Jean85** [comment]
```Generating autoload files
                                                   B>                           
                                                   B "Bp                        
.pp..                                              B    9p                      
 "9BBBBBBpp.                                       B      9p                    
    " ""9BBBBBBpp                          .<eeP"B B      .B b                  
           "SANDROBpp              .     B B     B B      )B B                  
              "BFRABBBB>  .<pe6P\B B     B B     B B      $  B     .e           
                 5NICOBBB B     ·B B     B B     B Bqp.  :B  B     $ 4BBpp      
                   BMIKIB B        B     B B     B B   "^Bp  B    ) |BBB"\BBpp. 
                 .BALEBBB """9q.   B"""""B B"""""B B      1p B""""9p BBBBbBBBBBBB
               <BLUCABBBB B    "B  B     B B     B B       B B     9 9BBB< ^P"  
            .6BSERGIOBBBB B666666B B     B B     B B       9 P      7 9BBBBP    

PARAUNIT TEST 0.4 -- by Shark Dev Team @ Facile.it
................................................................................
................................................................................
................................................................................
................................................................................
................................................................................
................................................................................
................................................................................
...............................................................

Execution time -- 00:00:01 

Executed: 19 test classes, 623 tests
```

Added executed tests count. Yay!
